lib/orbit-common/cache/observables/has-one-observable.js

```diff
@@ -0,0 +1,78 @@
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/operator/last';
+import 'rxjs/add/operator/scan';
+import 'rxjs/add/operator/startWith';
+import 'rxjs/add/observable/concat';
+import 'rxjs/add/observable/empty';
+import 'orbit-common/rxjs/operators/matching';
+import RecordsObservable from './records-observable';
+
+function extractRecordFromHasOneResult(value) {
+  if (!value) { return null; }
+
+  const id = Object.keys(value)[0];
+  return value[id];
+}
+
+export default class HasOneObservable extends Observable {
+  constructor(subscribe, cache, record, relationship) {
+    super(subscribe);
+    this.cache = cache;
+    this.record = record;
+    this.relationship = relationship;
+  }
+
```

Comment:
The `lift` function is a hook that's used when an operator returns a new instance of the class. In RxJS 4 operators would always return `Observable`, in RxJS 5 the `lift` hook makes it possible to return subclasses. The integration feels a little messy at the moment, it's on my list for Tuesday.